test/test_tensorboard.py

```diff
@@ -0,0 +1,454 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+from common_utils import run_tests
+
+from torch.utils.tensorboard import x2num, SummaryWriter
+import torch
+import numpy as np
+import unittest
+from torch.utils.tensorboard import summary
+from torch.utils.tensorboard.utils import _prepare_video, convert_to_HWC
+import pytest
+# from torch.utils.tensorboard import SummaryWriter
+# from .expect_reader import compare_proto
+import os
+import sys
+import matplotlib.pyplot as plt
+# from torch.utils.tensorboard import x2num
+
+
+class PyTorchNumpyTest(unittest.TestCase):
+    def test_pytorch_np(self):
+        tensors = [torch.rand(3, 10, 10), torch.rand(1), torch.rand(1, 2, 3, 4, 5)]
+        for tensor in tensors:
+            # regular tensor
+            assert isinstance(x2num.make_np(tensor), np.ndarray)
+
+            # CUDA tensor
+            if torch.cuda.device_count() > 0:
+                assert isinstance(x2num.make_np(tensor.cuda()), np.ndarray)
+
+            # regular variable
+            assert isinstance(x2num.make_np(torch.autograd.Variable(tensor)), np.ndarray)
+
+            # CUDA variable
+            if torch.cuda.device_count() > 0:
+                assert isinstance(x2num.make_np(torch.autograd.Variable(tensor).cuda()), np.ndarray)
+
+        # python primitive type
+        assert(isinstance(x2num.make_np(0), np.ndarray))
+        assert(isinstance(x2num.make_np(0.1), np.ndarray))
+
+    def test_pytorch_write(self):
+        with SummaryWriter() as w:
+            w.add_scalar('scalar', torch.autograd.Variable(torch.rand(1)), 0)
```

Comment:
So, what's going on here? Are we just testing that SummaryWriter doesn't crash? Is there any better testing we can do? (I guess for a visualization library, this can be hard. Should we have a human-run smoketest instead?)

Reply:
Yes, this is a smoke test. I usually check the result visually after running pytest. On tensorboard side, there are tests that make sure the data written by the underlying file writer can be read back correctly by the file reader. Here, `compare_proto` does the similar test at a higher level.